cclib/parser/turbomoleparser.py

```diff
@@ -123,6 +123,34 @@ def extract(self, inputfile, line):
                 revision = tokens[2]
                 self.metadata["package_version"] = "{}.r{}".format(package_version, revision)
 
+
+        ## Basis set info from dscf.
+        #               +--------------------------------------------------+
+        #               |               basis set information              |
+        #               +--------------------------------------------------+
+        # 
+        #               we will work with the 1s 3p 5d 7f 9g ... basis set
+        #               ...i.e. with spherical basis functions...
+        # 
+        #    type   atoms  prim   cont   basis
+        #    ---------------------------------------------------------------------------
+        #     o        1     15      5   sto-3g hondo  [2s1p|6s3p]
+        #     h        2      3      1   sto-3g hondo  [1s|3s]
+        #    ---------------------------------------------------------------------------
+        if "type   atoms  prim   cont   basis" in line:
+            line = next(inputfile)
+            line = next(inputfile)
+            basis_sets = []
+            while "---------------------------------------------------------------------------" not in line:
+                basis_sets.append(" ".join(line.split()[4:-1]))
+                line = next(inputfile)
+
+            # Turbomole gives us the basis set for each atom, but we're only interested if the same basis set is used throughout (for now).
+            if len(set(basis_sets)) == 1:
+                self.metadata["basis_set"] = list(set(basis_sets))[0]

```

Comment:
Since there is no guarantee of set ordering, this will pick a random basis set name. Maybe if the size of the set it greater than one, log a warning? This should at least be made repeatable by using a sorted list.

Reply:
Sorry as I understand it if there is more than 1 basis set then won't this just not get set? 
To me this seems like the behavior we would want. If it couldn't be parsed correctly we just don't set it.

Reply:
Yes that is correct, at present the basis set will not be set if more than one is given.
Not sure what is done elsewhere, but I guess the alternative would be to set the most 'popular' basis set (the one used for the most atoms) instead?

Reply:
Oh sorry, my eyes completely glazed over the `if len(set(basis_sets)) == 1:` line. I like that solution (the current one) more than the "most popular" solution.